v3/lints/cabf_br/lint_subject_common_name_not_exactly_from_san_test.go

```diff
@@ -0,0 +1,81 @@
+package cabf_br
+
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"testing"
+
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/test"
+)
+
+func TestCnNotExactlyFromSAN(t *testing.T) {
+	var testCases = []struct {
+		name      string
+		inputFile string
+
+		expectedOutput lint.LintStatus
+	}{
+		{
+			name:           "Pass - commonName in SAN.DNSNames",
+			inputFile:      "SANWithCNSeptember2021.pem",
+			expectedOutput: lint.Pass,
+		},
+		{
+			name:           "Pass - common name and SAN.IPAddress, IPv4",
+			inputFile:      "SANIPv4Address.pem",
+			expectedOutput: lint.Pass,
+		},
+		{
+			name:           "Pass - common name and SAN.IPAddress, IPv6",
+			inputFile:      "SANIPv6Address.pem",
+			expectedOutput: lint.Pass,
+		},
+		{
+			name:           "Error - common name not in SAN.DNSNames",
+			inputFile:      "SANWithoutCNSeptember2021.pem",
+			expectedOutput: lint.Error,
+		},
+		{
+			name:           "Error - common name in SAN.DNSNames but case mismatch",
+			inputFile:      "SANCaseNotMatchingCNSeptember2021.pem",
+			expectedOutput: lint.Error,
+		},
+		{
+			name:           "Error - common name not in SAN.IPAddresses, IPv4",
+			inputFile:      "SANIPv4AddressNotMatchingCommonName.pem",
+			expectedOutput: lint.Error,
+		},
+		{
+			name:           "Error - common name not in SAN.IPAddresses, IPv6",
+			inputFile:      "SANIPv6AddressNotMatchingCommonName.pem",
+			expectedOutput: lint.Error,
+		},
+		{
+			name:           "NE - certificate issued before 21 August 2021",
+			inputFile:      "SANWithMissingCN.pem",
+			expectedOutput: lint.NE,
+		},
+	}
```

Comment:
Silly question, are both of those considered to be valid or is the latter invalid?  I've got test certs for both they are both accepted which I assume should be fine as in the latter case both IP addresses are equivalent but I wanted to check given the discussion on the BRs that kicked all this off about things being functionally equivalent but not exact copies.

Reply:
I'm going to answer my own question and say, "this probably needs work" as...
This: "2001:db8:0:0:1:0:0:1 in the SAN, and 2001:db8::1:0:0:1 in the CN" passes whereas
this: "2001:db8::1:0:0:1 in the SAN, and 2001:db8:0:0:1:0:0:1 in the CN" does not.
Clearly, if you put the minimal version in the CN it works because the IP.String() function abbreviates the first 0:0 pair in the SAN version as per RFC but if it's the other way round then the SAN will remain as the minimal version and the '==' check will fail as the CN is not the correct representation.